support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/serializer/JacksonBinarySerializer.java

```diff
@@ -0,0 +1,81 @@
+package org.apereo.cas.serializer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.apereo.cas.TicketSerializer;
+import org.apereo.cas.ticket.ServiceTicket;
+import org.apereo.cas.ticket.ServiceTicketImpl;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketGrantingTicket;
+import org.apereo.cas.ticket.TicketGrantingTicketImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * @author David Rodriguez
+ *
+ * @since 5.1.0
+ */
+public class JacksonBinarySerializer implements TicketSerializer<ByteBuffer> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JacksonBinarySerializer.class);
+    private ObjectMapper mapper;
+
+    public JacksonBinarySerializer() {
+        mapper = Jackson2ObjectMapperBuilder.json()
+                .featuresToDisable(DeserializationFeature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE)
+                .featuresToDisable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+                .build();
+        mapper.findAndRegisterModules();
+    }
+
+    @Override
+    public ByteBuffer serializeTGT(final Ticket ticket) {
+        byte[] serialized;
+        try {
+            serialized = mapper.writeValueAsBytes(ticket);
+        } catch (final JsonProcessingException e) {
+            LOGGER.info("Error writing ticket {}: {}", ticket.getId(), e);
+            serialized = new byte[]{};
+        }
+        return ByteBuffer.wrap(serialized);
+    }
+
+    @Override
+    public ByteBuffer serializeST(final Ticket ticket) {
+        byte[] serialized;
+        try {
+            serialized = mapper.writeValueAsBytes(ticket);
+        } catch (final JsonProcessingException e) {
+            LOGGER.info("Error writing ticket {}: {}", ticket.getId(), e);
+            serialized = new byte[]{};
+        }
+        return ByteBuffer.wrap(serialized);
+    }
+
+    @Override
+    public TicketGrantingTicket deserializeTGT(final ByteBuffer ticket) {
+        try {
+            return mapper.readValue(ticket.array(), TicketGrantingTicketImpl.class);
```

Comment:
I'd promised to provide some notes on this works:

The general idea is, modules that have extended the ticket hierarchy attempt to register their own special ticket types into the catalog. This means, no storage system should have to make assumptions about what is available statically. It will need to look into the catalog to figure out what is available and proceed based on those. For example, OAuth/OIDC/etc register special ticket types, and Hazelcast as an example splits/auto-creates caches for each if/when needed by simply looking up ticket definitions in the catalog. 

In your particular example, you want to try to do the same sort of thing. You don't have to create a "cache" for every ticket type. That's up to you. Not all registries today actually do this; for instance Ehcache stuff everything into a single cache (see https://github.com/apereo/cas/issues/2561). You as the designer of the registry must decide/design how to best scale and optimize the internal configuration/creation of all things registry/storage related. If you need one table per ticket, great. If you don't, also great. 

To recap, the key thing here is to simply not make assumptions.